### PR TITLE
CQDG-684 apply vqsr or hard filtering depending on sequencing type

### PR DESCRIPTION
**Pull request by LysianeBouchard**
The goal of this PR is to introduce logic to adapt filtering logic (vqsr vs hard filtering) depending on the sequencing data type (whole exome sequencing or whole genome sequencing).

It is assumed that the sequencing type is passed as an extra column in the sample file. We use the code `WES` for whole exome sequencing and `WGS` for whole genome sequencing. 

For compatibility, we support both this new format and the previous sample file format.  One can specify the desired format through configuration parameter "sampleFileFormat".  Use value `V1` for the previous format and `V2` for the new format.  By default, we assume the previous format, i.e. `V1`. 

If using the previous format (`V1`), by default, it will be assume that the data is whole genome sequencing data (`WGS`).  One can change this by setting the configuration parameter `sequencingType`. Use `WES` for whole exome sequencing or `WGS` for whole genome sequencing.

Note that the sequencingType option will be ignored when using the sample file format `V2`.

**Examples usage**

To use the new sample file format :
`nextlfow run main.nf -c nextflow.config [other options]  --sampleFileFormat=V2`

To use the previous sample file format and whole genome sequencing, run the nextflow command normally without specifying any extra option.

To use the previous sample file format and whole exome sequencing data:
`nextlfow run main.nf -c nextflow.config [other options]  --sequencingType=WES`

**Test procedures**

Used tiny datasets to test the following scenarios:

- successfully process whole genome sequencing data with sample file format `V2`
- successfully process whole genome sequencing data with sample file format `V1`
- successfully process whole exome sequencing data with sample file format `V2`
- successfully process whole exome sequencing data with sample file format `V1`

- make sure that, by default, sample file format is set to `V1` and sequencing type to `WGS`

- job configured for sample file format `V1`, but we pass a sample file with format `V2`: job fail early, within the first process
- job configured for sample file format `V2`, but we pass a sample file with format `V1`: job fail early, before the first process

- invalid sample file format:  job is cancelled with an informative message
- invalid sequencing type format: (when using format `V1` only): job is cancelled with an informative message

**Extra minor changes**

This PR also contains the following minor changes:
    - use only dedicated params TSfilterINDEL and TSfilterSNP in vqsr processes
    - add comments
    - minor cosmetic changes (ex: spacing)

### CODE REVIEW

File: modules/vqsr.nf [hunk 1]
```diff
@@ -107,12 +122,13 @@ process applyVQSRIndel {
     tuple val(prefix), path(recal), path(tranches), path(vcf)
 
     output:
-    tuple val(prefix), path("*.snpindel.vqsr_${params.TSfilter}.vcf.gz*")
```
Comment (LysianeBouchard) [inline modules/vqsr.nf:110]: In the script, the TSfilterINDEL parameter is used, so I thought that we should probably use it here as well?

File: modules/vqsr.nf [hunk 2]
```diff
@@ -77,7 +90,7 @@ process applyVQSRSNP {
     tuple val(prefix), path(recal), path(tranches), path(vcf)
 
     output:
-    tuple val(prefix), path("*.snp.vqsr_${params.TSfilter}.vcf.gz*")
```
Comment (LysianeBouchard) [inline modules/vqsr.nf:80]: In the script, the TSfilterSNP parameter is used, so I thought that we should probably use it here as well?

File: main.nf [hunk 3]
```diff
@@ -76,29 +74,62 @@ process genotypeGVCF {
...
+- size: number of patients in the family
+- sequencingType: type of sequencing data, can be either 'WGS' or 'WXS'
+
+The expected input tsv format is the following:
+familyId   sequencingType   filePatient1   filePatient2  ...
+
+Potiential improvements:
+- Consider switching to json format, which is more suitable for complex types such as lists.
+- Rethink propagation of metadata in the workflow: https://training.nextflow.io/advanced/metadata/
+- Add validation logic
```
Comment (LysianeBouchard) [inline main.nf:114]: Did not include it here, but, I think it could be good to support both the new format with the extra sequencingType column and the previous format. This way we can still run the pipeline on older data. 

I am not sure exactly how to implement this. Could be via a config parameters giving an input data format "version". We can use different parsing logic depending on the "version".  On the older data format, we could infer the sequencing type from a parameter.

File: modules/vqsr.nf (hunk 2 above)
Comment (LysianeBouchard) [reply modules/vqsr.nf:80]: Validated with @DamienGnst  that we should well use TSfilterSNP.

File: main.nf (hunk 3 above)
Comment (LysianeBouchard) [reply main.nf]: Added support for both previous and new format. See details in the PR description.

File: main.nf [hunk 4]
```diff
@@ -1,12 +1,155 @@
...
+
+In the V1 format, the sequencing type is assumed to be the same for all samples and is specified through
+the configuration parameter `sequencingType` (default to `WGS`).
+
+## Potiential improvements ##
+
+Consider the following improvements:
+- Rethink propagation of metadata in the workflow: https://training.nextflow.io/advanced/metadata/
+- Add validation step
+*/
```
Comment (LysianeBouchard) [inline main.nf:53]: Wish list from @DamienGnst  (to be addressed in another PR)
- Also add a subworkflow for vep
- Add restart point (see Sarek for an example: https://raw.githubusercontent.com/nf-core/sarek/3.4.1/docs/images/sarek_subway.png)

File: main.nf [hunk 5]
```diff
@@ -49,13 +194,11 @@ process importGVCF {
     gatk -version
     gatk --java-options "-Xmx8g"  CombineGVCFs -R $referenceGenome/${params.referenceGenomeFasta} $exactGvcfFiles -O ${familyId}.combined.gvcf.gz -L $broadResource/${params.intervalsFile}
```
Comment (LysianeBouchard) [inline main.nf:195]: Comment from @DamienGnst  (for another PR)
For options related to memory and cpu capacity such as  `--java-options "-Xmx8g"` above, we would like to infer the right value from  the configured memory/cpu on the process.  We might need to keep a bit of memory/cpu for fusion and nextflow internal logic ... Ex: use the configured  process memory - 1  etc.).

**Review by DamienGnst — APPROVED** [review]

**Comment by LysianeBouchard** [comment]
Noted proposed improvements here:
https://www.notion.so/ferlab/caf842d6368a4e1d8ef578f61ad1dac3?v=043134ab4f4e4ba69547f49d3ccda37c&p=15443e7ff0da4a75b062c4089d2ac67e&pm=s
